Fix Doxygen warnings

Thanks to the efforts of @richterjakob and @menon-karthik, we have very little undocumented code.

If we fix the last 60, we could make the Doxygen build fail on warnings to enforce that only documented new code can be merged. We would have to add `WARN_AS_ERROR = FAIL_ON_WARNINGS` to the Doxyfile (run through the whole build but throw an error if there's a warning).

The warnings are (mostly undocumented class variables):
```
Parsing file /Users/pfaller/work/r/Users/pfaller/work/repos/svZeroDPlus_fork/src/interface/interface.cpp:149: warning: documented symbol 'throw std::runtime_error' was not declared or defined.
src/interface/interface.cpp:153: warning: documented symbol 'throw std::runtime_error' was not declared or defined.
Building full member lists recursivel/Users/pfaller/work/repos/svZeroDPlus_fork/src/interface/interface.h:47: warning: Member S (typedef) of file interface.h is not documented.
src/model/ClosedLoopCoronaryBC.h:43: warning: Member Side (enumeration) of namespace zd_model is not documented.
src/model/ClosedLoopHeartPulmonary.h:41: warning: Member PI (macro definition) of file ClosedLoopHeartPulmonary.h is not documented.
src/optimize/calibrate.cpp:37: warning: Member calibrate(const nlohmann::json &config) (function) of namespace optimize is not documented.
src/solve/csv_writer.cpp:169: warning: The following parameter of io::to_variable_csv(std::vector< double > &times, std::vector< algebra::State > &states, zd_model::Model &model, bool mean, bool derivative) is not documented:
src/solve/csv_writer.cpp:38: warning: The following parameter of io::to_vessel_csv(std::vector< double > &times, std::vector< algebra::State > &states, zd_model::Model &model, bool mean, bool derivative) is not documented:
src/solve/debug.h:40: warning: Member DEBUG_MSG(str) (macro definition) of file debug.h is not documented.
src/interface/interface.h:55: warning: Member SolverInterface(const std::string &input_file_name) (function) of class SolverInterface is not documented.
src/interface/interface.h:61: warning: Member problem_id_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:62: warning: Member input_file_name_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:66: warning: Member external_step_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:69: warning: Member time_step_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:70: warning: Member num_time_steps_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:71: warning: Member absolute_tolerance_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:72: warning: Member max_nliter_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:73: warning: Member time_step_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:74: warning: Member save_interval_counter_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:75: warning: Member output_interval_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:76: warning: Member system_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:77: warning: Member num_output_steps_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:78: warning: Member pts_per_cycle_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:79: warning: Member output_last_cycle_only_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:81: warning: Member model_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:82: warning: Member integrator_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:84: warning: Member state_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:85: warning: Member times_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:86: warning: Member states_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:58: warning: Member problem_id_count_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:59: warning: Member interface_list_ (variable) of class SolverInterface is not documented.
Generating docs for compound optimize::LevenbergMarquardtOptimizer.../Users/pfaller/work/repos/svZeroDPlus_fork/src/model/Block.h:225: warning: return type of member zd_model::Block::get_num_triplets is not documented
src/model/Block.h:131: warning: argument 'num_internal_vars' of command @param is not found in the argument list of zd_model::Block::setup_dofs_(DOFHandler &dofhandler, unsigned int num_equations, std::list< std::string > internal_var_names)
src/model/Block.h:131: warning: The following parameter of zd_model::Block::setup_dofs_(DOFHandler &dofhandler, unsigned int num_equations, std::list< std::string > internal_var_names) is not documented:
src/model/BloodVessel.h:147: warning: Member BloodVessel(int id, const std::vector< int > &param_ids, Model *model) (function) of class zd_model::BloodVessel is not documented.
src/model/BloodVessel.h:103: warning: unable to resolve reference to 'Parameters::stenosis_coefficient' for \ref command
src/model/BloodVessel.h:216: warning: return type of member zd_model::BloodVessel::get_num_triplets is not documented
src/model/BloodVesselJunction.h:214: warning: return type of member zd_model::BloodVesselJunction::get_num_triplets is not documented
src/model/ClosedLoopCoronaryBC.h:123: warning: Member ClosedLoopCoronaryBC(int id, const std::vector< int > &param_ids, Model *model, Side side) (function) of class zd_model::ClosedLoopCoronaryBC is not documented.
src/model/ClosedLoopCoronaryBC.h:189: warning: return type of member zd_model::ClosedLoopCoronaryBC::get_num_triplets is not documented
src/model/ClosedLoopHeartPulmonary.h:191: warning: return type of member zd_model::ClosedLoopHeartPulmonary::get_num_triplets is not documented
src/model/ClosedLoopRCRBC.h:165: warning: return type of member zd_model::ClosedLoopRCRBC::get_num_triplets is not documented
src/model/FlowReferenceBC.h:134: warning: return type of member zd_model::FlowReferenceBC::get_num_triplets is not documented
src/model/Junction.h:149: warning: Member num_inlets (variable) of class zd_model::Junction is not documented.
src/model/Junction.h:150: warning: Member num_outlets (variable) of class zd_model::Junction is not documented.
src/model/Junction.h:144: warning: return type of member zd_model::Junction::get_num_triplets is not documented
src/model/Model.h:137: warning: The following parameter of zd_model::Model::add_node(const std::vector< Block * > &inlet_eles, const std::vector< Block * > &outlet_eles, std::string_view name) is not documented:
src/model/Model.h:121: warning: argument 'block_id' of command @param is not found in the argument list of zd_model::Model::get_block_type(std::string_view name)
src/model/Model.h:121: warning: The following parameter of zd_model::Model::get_block_type(std::string_view name) is not documented:
src/model/Model.h:255: warning: parameters of member zd_model::Model::get_num_blocks are not documented
src/model/Node.h:66: warning: Member Node(int id, const std::vector< Block * > &inlet_eles, const std::vector< Block * > &outlet_eles, Model *model) (function) of class zd_model::Node is not documented.
src/model/Node.h:60: warning: zd_model::Node::Node has @param documentation sections but no arguments
src/model/Node.h:93: warning: unable to resolve reference to 'global_var_ids' for \ref command
src/model/Node.h:93: warning: unable to resolve reference to 'global_eqn_ids' for \ref command
src/model/OpenLoopCoronaryBC.h:160: warning: return type of member zd_model::OpenLoopCoronaryBC::get_num_triplets is not documented
src/model/Parameter.h:66: warning: The following parameter of zd_model::Parameter::Parameter(int id, const std::vector< double > &times, const std::vector< double > &values, bool periodic=true) is not documented:
src/model/PressureReferenceBC.h:135: warning: return type of member zd_model::PressureReferenceBC::get_num_triplets is not documented
src/model/ResistanceBC.h:133: warning: return type of member zd_model::ResistanceBC::get_num_triplets is not documented
src/model/ResistiveJunction.h:145: warning: return type of member zd_model::ResistiveJunction::get_num_triplets is not documented
src/model/WindkesselBC.h:157: warning: return type of member zd_model::WindkesselBC::get_num_triplets is not documented
```

--- a/.github/workflows/documentation.yml
+++ b/.github/workflows/documentation.yml
@@ -12,7 +12,6 @@ jobs:
         run: mkdir docs/build
       - name: Build doxygen documentation
         continue-on-error: false
-        #uses: mattnotmitt/doxygen-action@v1.9.5
         uses: mattnotmitt/doxygen-action@edge
         with:
           working-directory: '.'

--- a/.github/workflows/documentation.yml
+++ b/.github/workflows/documentation.yml
@@ -11,7 +11,7 @@ jobs:
       - name: Make build directory
         run: mkdir docs/build
       - name: Build doxygen documentation
-        continue-on-error: false
+        continue-on-error: true
         uses: mattnotmitt/doxygen-action@edge
         with:
           working-directory: '.'

--- a/.github/workflows/documentation.yml
+++ b/.github/workflows/documentation.yml
@@ -11,7 +11,7 @@ jobs:
       - name: Make build directory
         run: mkdir docs/build
       - name: Build doxygen documentation
-        continue-on-error: true
+        continue-on-error: false
         uses: mattnotmitt/doxygen-action@edge
         with:
           working-directory: '.'

--- a/.github/workflows/documentation.yml
+++ b/.github/workflows/documentation.yml
@@ -12,7 +12,8 @@ jobs:
         run: mkdir docs/build
       - name: Build doxygen documentation
         continue-on-error: false
-        uses: mattnotmitt/doxygen-action@v1.9.5
+        #uses: mattnotmitt/doxygen-action@v1.9.5
+        uses: mattnotmitt/doxygen-action@edge
         with:
           working-directory: '.'
           doxyfile-path: 'docs/Doxyfile'

--- a/.github/workflows/documentation.yml
+++ b/.github/workflows/documentation.yml
@@ -12,7 +12,7 @@ jobs:
         run: mkdir docs/build
       - name: Build doxygen documentation
         continue-on-error: false
-        uses: mattnotmitt/doxygen-action@edge
+        uses: mattnotmitt/doxygen-action@v1.9.5
         with:
           working-directory: '.'
           doxyfile-path: 'docs/Doxyfile'

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -102,8 +102,8 @@ WARN_IF_DOC_ERROR      = YES
 WARN_NO_PARAMDOC       = YES
 WARN_FORMAT            = "$file:$line: $text"
 WARN_LOGFILE           =
-#WARN_AS_ERROR          = FAIL_ON_WARNINGS
-WARN_AS_ERROR          = YES
+WARN_AS_ERROR          = FAIL_ON_WARNINGS
+#WARN_AS_ERROR          = YES
 #---------------------------------------------------------------------------
 # Configuration options related to the input files
 #---------------------------------------------------------------------------

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -102,7 +102,8 @@ WARN_IF_DOC_ERROR      = YES
 WARN_NO_PARAMDOC       = YES
 WARN_FORMAT            = "$file:$line: $text"
 WARN_LOGFILE           =
-WARN_AS_ERROR          = FAIL_ON_WARNINGS
+#WARN_AS_ERROR          = FAIL_ON_WARNINGS
+WARN_AS_ERROR          = YES
 #---------------------------------------------------------------------------
 # Configuration options related to the input files
 #---------------------------------------------------------------------------

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -102,7 +102,6 @@ WARN_IF_DOC_ERROR      = YES
 WARN_NO_PARAMDOC       = YES
 WARN_FORMAT            = "$file:$line: $text"
 WARN_LOGFILE           =
-#WARN_AS_ERROR          = FAIL_ON_WARNINGS
 WARN_AS_ERROR          = YES
 #---------------------------------------------------------------------------
 # Configuration options related to the input files

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -102,6 +102,7 @@ WARN_IF_DOC_ERROR      = YES
 WARN_NO_PARAMDOC       = YES
 WARN_FORMAT            = "$file:$line: $text"
 WARN_LOGFILE           =
+WARN_AS_ERROR          = FAIL_ON_WARNINGS
 #---------------------------------------------------------------------------
 # Configuration options related to the input files
 #---------------------------------------------------------------------------

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -216,7 +216,8 @@ EXTRA_PACKAGES         = {tikz} \
                          {circuitikz} \
                          {amsmath} \
                          {amsfonts} \
-                         {amssymb}
+                         {amssymb} \
+                         [utf8]{inputenc}
 LATEX_HEADER           =
 LATEX_FOOTER           =
 LATEX_EXTRA_FILES      =

--- a/docs/Doxyfile
+++ b/docs/Doxyfile
@@ -102,8 +102,8 @@ WARN_IF_DOC_ERROR      = YES
 WARN_NO_PARAMDOC       = YES
 WARN_FORMAT            = "$file:$line: $text"
 WARN_LOGFILE           =
-WARN_AS_ERROR          = FAIL_ON_WARNINGS
-#WARN_AS_ERROR          = YES
+#WARN_AS_ERROR          = FAIL_ON_WARNINGS
+WARN_AS_ERROR          = YES
 #---------------------------------------------------------------------------
 # Configuration options related to the input files
 #---------------------------------------------------------------------------

--- a/docs/pages/developer_guide.md
+++ b/docs/pages/developer_guide.md
@@ -60,6 +60,13 @@ it with:
 make codecheck
 ```
 
+If the above commands do not work on your platform (it does not work on Sherlock at Stanford)
+you can run the following command **from the svZeroDPlus folder** to format all your files:
+
+```bash
+find src/**/*.h src/**/*.cpp | xargs clang-format -style=Google -i
+```
+
 The latter check is also performed in the GitHub CI/CD (a.k.a. Actions) and
 indicates on merge requests when the code doesnt yet meet all style
 requirements.

--- a/docs/pages/developer_guide.md
+++ b/docs/pages/developer_guide.md
@@ -47,7 +47,7 @@ There are tools for your favorite IDE to automatically format your code. Example
 - [and many more](https://clang.llvm.org/docs/ClangFormat.html)
 
 Before committing any changes, you can run the following
-comment **from your build folder** to format all your files:
+command **from your build folder** to format all your files:
 
 ```bash
 make codeformat

--- a/docs/pages/developer_guide.md
+++ b/docs/pages/developer_guide.md
@@ -74,11 +74,8 @@ for an introduction into the syntax of writing documentation in C++. For more
 inspiration, you can look at the existing source files and how they use
 documentation. 
 
-> **&#9432;** Note that undocumented code will fail the automated code checks 
-on Github and cannot be merged.
-
-**Note that undocumented code will fail the automated code checks on Github 
-and cannot be merged.**
+==**Note that undocumented code will fail the automated code checks on Github 
+and cannot be merged.**==
 
 In the following you can find a short recap of the most important
 commands:

--- a/docs/pages/developer_guide.md
+++ b/docs/pages/developer_guide.md
@@ -74,8 +74,8 @@ for an introduction into the syntax of writing documentation in C++. For more
 inspiration, you can look at the existing source files and how they use
 documentation. 
 
-==**Note that undocumented code will fail the automated code checks on Github 
-and cannot be merged.**==
+**NOTE: Undocumented code will fail the automated code checks on Github 
+and cannot be merged.**
 
 In the following you can find a short recap of the most important
 commands:

--- a/docs/pages/developer_guide.md
+++ b/docs/pages/developer_guide.md
@@ -72,7 +72,15 @@ We use [Doxygen](https://doxygen.nl) to automatically build an html documentatio
 from source code. Please have at Doxygen's [Documentation Guide](https://www.doxygen.nl/manual/docblocks.html)
 for an introduction into the syntax of writing documentation in C++. For more
 inspiration, you can look at the existing source files and how they use
-documentation. In the following you can find a short recap of the most important
+documentation. 
+
+> **&#9432;** Note that undocumented code will fail the automated code checks 
+on Github and cannot be merged.
+
+**Note that undocumented code will fail the automated code checks on Github 
+and cannot be merged.**
+
+In the following you can find a short recap of the most important
 commands:
 
 ### Latex equations

--- a/docs/references.bib
+++ b/docs/references.bib
@@ -8,7 +8,6 @@ year = {2000},
 issn = {0045-7825},
 url = {https://doi.org/10.1016/S0045-7825(00)00203-6},
 author = {Kenneth E. Jansen and Christian H. Whiting and Gregory M. Hulbert},
-abstract = {A generalized-α method is developed and analyzed for linear, first-order systems. The method is then extended to the filtered Navier–Stokes equations within the context of a stabilized finite element method. The formulation is studied through the application to laminar flow past a circular cylinder and turbulent flow past a long, transverse groove. The method is formulated to obtain a second-order accurate family of time integrators whose high frequency amplification factor is the sole free parameter. Such an approach allows the replication of midpoint rule (zero damping), Gear's method (maximal damping), or anything in between.}
 }
 
 @article{kim_coronary,

--- a/docs/references.bib
+++ b/docs/references.bib
@@ -1,5 +1,5 @@
 @article{JANSEN2000305,
-title = {A generalized alpha method for integrating the filtered Navier–Stokes equations with a stabilized finite element method},
+title = {A generalized-alpha method for integrating the filtered Navier–Stokes equations with a stabilized finite element method},
 journal = {Computer Methods in Applied Mechanics and Engineering},
 volume = {190},
 number = {3},

--- a/docs/references.bib
+++ b/docs/references.bib
@@ -1,5 +1,5 @@
 @article{JANSEN2000305,
-title = {A generalized-α method for integrating the filtered Navier–Stokes equations with a stabilized finite element method},
+title = {A generalized alpha method for integrating the filtered Navier–Stokes equations with a stabilized finite element method},
 journal = {Computer Methods in Applied Mechanics and Engineering},
 volume = {190},
 number = {3},

--- a/src/algebra/Integrator.h
+++ b/src/algebra/Integrator.h
@@ -178,6 +178,8 @@ class Integrator {
   /**
    * @brief Get average number of nonlinear iterations in all step calls
    *
+   * @return Average number of nonlinear iterations in all step calls
+   *
    */
   double avg_nonlin_iter();
 };

--- a/src/interface/interface.cpp
+++ b/src/interface/interface.cpp
@@ -118,7 +118,6 @@ void initialize(std::string input_file_arg, int& problem_id, int& pts_per_cycle,
   auto simparams = load_simulation_params(config);
 
   auto model = std::shared_ptr<Model>(new Model());
-  // auto model = std::shared_ptr<MODEL::Model<T>>(new MODEL::Model<T>());
 
   load_simulation_model(config, *model.get());
   auto state = load_initial_condition(config, *model.get());
@@ -163,7 +162,7 @@ void initialize(std::string input_file_arg, int& problem_id, int& pts_per_cycle,
   interface->absolute_tolerance_ = simparams.sim_abs_tol;
   interface->time_step_ = 0;
   interface->system_size_ = model->dofhandler.size();
-  interface->output_interval_ = simparams.output_interval;
+  //interface->output_interval_ = simparams.output_interval;
   interface->num_time_steps_ = simparams.sim_num_time_steps;
   interface->pts_per_cycle_ = simparams.sim_pts_per_cycle;
   pts_per_cycle = simparams.sim_pts_per_cycle;

--- a/src/interface/interface.cpp
+++ b/src/interface/interface.cpp
@@ -34,7 +34,7 @@
 
 #include "SimulationParameters.h"
 
-using S = SparseSystem;
+//using S = SparseSystem;
 
 // Static member data.
 int SolverInterface::problem_id_count_ = 0;

--- a/src/interface/interface.cpp
+++ b/src/interface/interface.cpp
@@ -34,7 +34,7 @@
 
 #include "SimulationParameters.h"
 
-//using S = SparseSystem;
+// using S = SparseSystem;
 
 // Static member data.
 int SolverInterface::problem_id_count_ = 0;
@@ -162,7 +162,7 @@ void initialize(std::string input_file_arg, int& problem_id, int& pts_per_cycle,
   interface->absolute_tolerance_ = simparams.sim_abs_tol;
   interface->time_step_ = 0;
   interface->system_size_ = model->dofhandler.size();
-  //interface->output_interval_ = simparams.output_interval;
+  // interface->output_interval_ = simparams.output_interval;
   interface->num_time_steps_ = simparams.sim_num_time_steps;
   interface->pts_per_cycle_ = simparams.sim_pts_per_cycle;
   pts_per_cycle = simparams.sim_pts_per_cycle;

--- a/src/interface/interface.cpp
+++ b/src/interface/interface.cpp
@@ -34,8 +34,6 @@
 
 #include "SimulationParameters.h"
 
-// using S = SparseSystem;
-
 // Static member data.
 int SolverInterface::problem_id_count_ = 0;
 std::map<int, SolverInterface*> SolverInterface::interface_list_;
@@ -162,7 +160,6 @@ void initialize(std::string input_file_arg, int& problem_id, int& pts_per_cycle,
   interface->absolute_tolerance_ = simparams.sim_abs_tol;
   interface->time_step_ = 0;
   interface->system_size_ = model->dofhandler.size();
-  // interface->output_interval_ = simparams.output_interval;
   interface->num_time_steps_ = simparams.sim_num_time_steps;
   interface->pts_per_cycle_ = simparams.sim_pts_per_cycle;
   pts_per_cycle = simparams.sim_pts_per_cycle;

--- a/src/interface/interface.h
+++ b/src/interface/interface.h
@@ -108,11 +108,6 @@ class SolverInterface {
    */
   int time_step_ = 0.0;
   /**
-   * @brief Current time step
-   */
-  // int save_interval_counter_ = 0;
-  // int output_interval_ = 0;
-  /**
    * @brief The size of the 0D system
    */
   int system_size_ = 0;

--- a/src/interface/interface.h
+++ b/src/interface/interface.h
@@ -44,8 +44,6 @@
 #include "csv_writer.h"
 #include "debug.h"
 
-//using S = SparseSystem;
-
 /**
  * @brief Interface class for calling svZeroD from external programs
  */
@@ -56,7 +54,7 @@ class SolverInterface {
    * @param input_file_name The 0D JSON file which specifies the model
    */
   SolverInterface(const std::string& input_file_name);
-  
+
   /**
    * @brief Destroy the interface object
    */
@@ -75,7 +73,7 @@ class SolverInterface {
    * @brief ID of current interface
    */
   int problem_id_ = 0;
-  
+
   /**
    * @brief 0D input (JSON) file
    */
@@ -112,8 +110,8 @@ class SolverInterface {
   /**
    * @brief Current time step
    */
-  //int save_interval_counter_ = 0;
-  //int output_interval_ = 0;
+  // int save_interval_counter_ = 0;
+  // int output_interval_ = 0;
   /**
    * @brief The size of the 0D system
    */

--- a/src/interface/interface.h
+++ b/src/interface/interface.h
@@ -44,7 +44,7 @@
 #include "csv_writer.h"
 #include "debug.h"
 
-using S = SparseSystem;
+//using S = SparseSystem;
 
 /**
  * @brief Interface class for calling svZeroD from external programs

--- a/src/interface/interface.h
+++ b/src/interface/interface.h
@@ -49,39 +49,107 @@ using S = SparseSystem;
 /**
  * @brief Interface class for calling svZeroD from external programs
  */
-
 class SolverInterface {
  public:
+  /**
+   * @brief Construct a new interface object
+   * @param input_file_name The 0D JSON file which specifies the model
+   */
   SolverInterface(const std::string& input_file_name);
+  
+  /**
+   * @brief Destroy the interface object
+   */
   ~SolverInterface();
 
+  /**
+   * @brief Counter for the number of interfaces
+   */
   static int problem_id_count_;
+  /**
+   * @brief List of interfaces
+   */
   static std::map<int, SolverInterface*> interface_list_;
 
+  /**
+   * @brief ID of current interface
+   */
   int problem_id_ = 0;
+  
+  /**
+   * @brief 0D input (JSON) file
+   */
   std::string input_file_name_;
 
-  // Parameters for the external solver (the calling program).
-  // This is set by the external solver via the interface.
+  /**
+   * @brief Time step size of the external program
+   *
+   * This is required for coupling with a 3D solver
+   */
   double external_step_size_ = 0.1;
 
   // These are read in from the input JSON solver configuration file.
+  /**
+   * @brief 0D time step size
+   */
   double time_step_size_ = 0.0;
+  /**
+   * @brief Number of 0D time steps
+   */
   int num_time_steps_ = 0;
+  /**
+   * @brief Convergence tolerance for the 0D model
+   */
   double absolute_tolerance_ = 0.0;
+  /**
+   * @brief Maximum number of non-linear iterations
+   */
   int max_nliter_ = 0;
+  /**
+   * @brief Current time step
+   */
   int time_step_ = 0.0;
-  int save_interval_counter_ = 0;
-  int output_interval_ = 0;
+  /**
+   * @brief Current time step
+   */
+  //int save_interval_counter_ = 0;
+  //int output_interval_ = 0;
+  /**
+   * @brief The size of the 0D system
+   */
   int system_size_ = 0;
+  /**
+   * @brief The number of steps to output
+   */
   int num_output_steps_ = 0;
+  /**
+   * @brief Number of time steps per cycle
+   */
   int pts_per_cycle_ = 0;
+  /**
+   * @brief Output results from last cycle only?
+   */
   bool output_last_cycle_only_ = false;
 
+  /**
+   * @brief The current 0D model object
+   */
   std::shared_ptr<Model> model_;
+  /**
+   * @brief The current 0D integrator object
+   */
   Integrator integrator_;
 
+  /**
+   * @brief The current 0D state vector
+   */
   State state_;
+  /**
+   * @brief Vector to store solution times
+   */
   std::vector<double> times_;
+  /**
+   * @brief Vector to store solution states
+   */
   std::vector<State> states_;
 };

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -49,7 +49,7 @@
 struct TripletsContributions {
   TripletsContributions(){};
   /**
-   * @brief The number of triplets that the element contributes
+   * @brief Set the number of triplets that the element contributes
    * to the global system.
    * @param F Contributions to F matrix
    * @param E Contributions to E matrix
@@ -57,10 +57,11 @@ struct TripletsContributions {
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   /**
-   * @brief The number of triplets that the element contributes
+   * @brief Set the number of triplets that the element contributes
    * to the global system.
-   * @param other TripletsContributions object to add to the 
+   * @param other TripletsContributions object to add to the
    * number of contributions
+   * @return The number of triplets
    */
   TripletsContributions operator+=(const TripletsContributions &other) {
     F += other.F;

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -53,6 +53,7 @@ struct TripletsContributions {
    * to the global system.
    * @param F Contributions to F matrix
    * @param E Contributions to E matrix
+   * @param D Contributions to dC/dy matrix
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   /**
@@ -78,7 +79,7 @@ struct TripletsContributions {
    */
   int E{0};
   /**
-   * @brief Contributions to D matrix
+   * @brief Contributions to dC/dy matrix
    */
   int D{0};
 };

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -53,7 +53,6 @@ struct TripletsContributions {
    * to the global system.
    * @param F Contributions to F matrix
    * @param E Contributions to E matrix
-   * @param D Contributions to D matrix
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   /**

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -53,6 +53,7 @@ struct TripletsContributions {
    * to the global system.
    * @param F Contributions to F matrix
    * @param E Contributions to E matrix
+   * @param D Contributions to E matrix
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   /**

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -53,7 +53,6 @@ struct TripletsContributions {
    * to the global system.
    * @param F Contributions to F matrix
    * @param E Contributions to E matrix
-   * @param D Contributions to E matrix
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   /**

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -152,7 +152,7 @@ class Block {
    * @param dofhandler Degree-of-freedom handler to register variables and
    * equations at
    * @param num_equations Number of equations of the block
-   * @param num_internal_vars Number of internal variables of the block
+   * @param internal_var_names Number of internal variables of the block
    */
 
   void setup_dofs_(DOFHandler &dofhandler, int num_equations,

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -56,6 +56,12 @@ struct TripletsContributions {
    * @param D Contributions to D matrix
    */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
+  /**
+   * @brief The number of triplets that the element contributes
+   * to the global system.
+   * @param other TripletsContributions object to add to the 
+   * number of contributions
+   */
   TripletsContributions operator+=(const TripletsContributions &other) {
     F += other.F;
     E += other.E;

--- a/src/model/Block.h
+++ b/src/model/Block.h
@@ -48,6 +48,13 @@
  */
 struct TripletsContributions {
   TripletsContributions(){};
+  /**
+   * @brief The number of triplets that the element contributes
+   * to the global system.
+   * @param F Contributions to F matrix
+   * @param E Contributions to E matrix
+   * @param D Contributions to D matrix
+   */
   TripletsContributions(int F, int E, int D) : F(F), E(E), D(D){};
   TripletsContributions operator+=(const TripletsContributions &other) {
     F += other.F;
@@ -56,8 +63,17 @@ struct TripletsContributions {
     return *this;
   };
 
+  /**
+   * @brief Contributions to F matrix
+   */
   int F{0};
+  /**
+   * @brief Contributions to E matrix
+   */
   int E{0};
+  /**
+   * @brief Contributions to D matrix
+   */
   int D{0};
 };
 
@@ -237,6 +253,8 @@ class Block {
    *
    * Number of triplets that the element contributes to the global system
    * (relevant for sparse memory reservation)
+   *
+   * @return TripletsContributions Number of triplets of element
    */
   virtual TripletsContributions get_num_triplets();
 };

--- a/src/model/BloodVessel.h
+++ b/src/model/BloodVessel.h
@@ -115,9 +115,8 @@
  * \f]
  *
  * with the stenosis resistance \f$ S=K_{t} \frac{\rho}{2
- * A_{o}^{2}}\left(\frac{A_{o}}{A_{s}}-1\right)^{2} \f$. The
- * constant part of the equation is summarized in \ref
- * Parameters::stenosis_coefficient. \f$R\f$, \f$C\f$, and \f$L\f$ refer to
+ * A_{o}^{2}}\left(\frac{A_{o}}{A_{s}}-1\right)^{2} \f$.
+ * \f$R\f$, \f$C\f$, and \f$L\f$ refer to
  * Poisieuille resistance, capacitance and inductance, respectively.
  *
  * ### Gradient

--- a/src/model/BloodVessel.h
+++ b/src/model/BloodVessel.h
@@ -154,6 +154,13 @@ class BloodVessel : public Block {
     STENOSIS_COEFFICIENT = 3,
   };
 
+  /**
+   * @brief Construct a new BloodVessel object
+   *
+   * @param id Global ID of the block
+   * @param param_ids Global IDs of the block parameters
+   * @param model The model to which the block belongs
+   */
   explicit BloodVessel(int id, const std::vector<int> &param_ids, Model *model)
       : Block(id, param_ids, model){};
 

--- a/src/model/ClosedLoopCoronaryBC.h
+++ b/src/model/ClosedLoopCoronaryBC.h
@@ -38,6 +38,10 @@
 #include "ClosedLoopHeartPulmonary.h"
 #include "SparseSystem.h"
 
+/**
+ * @brief Specify whether this BC is on a left or right coronary artery.
+ *
+ */
 enum class Side { LEFT, RIGHT, NONE };
 
 /**
@@ -118,6 +122,14 @@ enum class Side { LEFT, RIGHT, NONE };
  */
 class ClosedLoopCoronaryBC : public Block {
  public:
+  /**
+   * @brief Construct a ClosedLoopCoronaryBC object.
+   *
+   * @param id Global ID of the block
+   * @param param_ids Global IDs of the block parameters
+   * @param model The model to which the block belongs
+   * @param side Specify if this is a left or right coronary artery
+   */
   explicit ClosedLoopCoronaryBC(int id, const std::vector<int> &param_ids,
                                 Model *model, Side side)
       : Block(id, param_ids, model), side{side} {};

--- a/src/model/ClosedLoopHeartPulmonary.cpp
+++ b/src/model/ClosedLoopHeartPulmonary.cpp
@@ -186,12 +186,12 @@ void ClosedLoopHeartPulmonary::get_activation_and_elastance_functions(
   // Activation function
   AA = 0.0;
   if (t_in_cycle <= tpwave) {
-    AA = (0.5) * (1.0 - cos(2.0 * PI * (t_in_cycle - tpwave + Tsa) / Tsa));
+    AA = (0.5) * (1.0 - cos(2.0 * M_PI * (t_in_cycle - tpwave + Tsa) / Tsa));
   } else if ((t_in_cycle >= (T_cardiac - Tsa) + tpwave) and
              (t_in_cycle < T_cardiac)) {
     AA =
         (0.5) *
-        (1.0 - cos(2.0 * PI * (t_in_cycle - tpwave - (T_cardiac - Tsa)) / Tsa));
+        (1.0 - cos(2.0 * M_PI * (t_in_cycle - tpwave - (T_cardiac - Tsa)) / Tsa));
   } else {
     AA = 0.0;
   }
@@ -218,8 +218,8 @@ void ClosedLoopHeartPulmonary::get_activation_and_elastance_functions(
 
   for (auto i = 0; i < num_elast_modes; i++) {
     Elv_i = Elv_i +
-            (Ft_elastance[i][0]) * cos(2.0 * PI * i * t_in_cycle / T_cardiac) -
-            (Ft_elastance[i][1]) * sin(2.0 * PI * i * t_in_cycle / T_cardiac);
+            (Ft_elastance[i][0]) * cos(2.0 * M_PI * i * t_in_cycle / T_cardiac) -
+            (Ft_elastance[i][1]) * sin(2.0 * M_PI * i * t_in_cycle / T_cardiac);
   }
 
   Elv = Elv_i * parameters[global_param_ids[ParamId::ELV_S]];

--- a/src/model/ClosedLoopHeartPulmonary.cpp
+++ b/src/model/ClosedLoopHeartPulmonary.cpp
@@ -189,9 +189,8 @@ void ClosedLoopHeartPulmonary::get_activation_and_elastance_functions(
     AA = (0.5) * (1.0 - cos(2.0 * M_PI * (t_in_cycle - tpwave + Tsa) / Tsa));
   } else if ((t_in_cycle >= (T_cardiac - Tsa) + tpwave) and
              (t_in_cycle < T_cardiac)) {
-    AA =
-        (0.5) *
-        (1.0 - cos(2.0 * M_PI * (t_in_cycle - tpwave - (T_cardiac - Tsa)) / Tsa));
+    AA = (0.5) * (1.0 - cos(2.0 * M_PI *
+                            (t_in_cycle - tpwave - (T_cardiac - Tsa)) / Tsa));
   } else {
     AA = 0.0;
   }
@@ -217,9 +216,10 @@ void ClosedLoopHeartPulmonary::get_activation_and_elastance_functions(
   double Elv_i = 0.0;
 
   for (auto i = 0; i < num_elast_modes; i++) {
-    Elv_i = Elv_i +
-            (Ft_elastance[i][0]) * cos(2.0 * M_PI * i * t_in_cycle / T_cardiac) -
-            (Ft_elastance[i][1]) * sin(2.0 * M_PI * i * t_in_cycle / T_cardiac);
+    Elv_i =
+        Elv_i +
+        (Ft_elastance[i][0]) * cos(2.0 * M_PI * i * t_in_cycle / T_cardiac) -
+        (Ft_elastance[i][1]) * sin(2.0 * M_PI * i * t_in_cycle / T_cardiac);
   }
 
   Elv = Elv_i * parameters[global_param_ids[ParamId::ELV_S]];

--- a/src/model/ClosedLoopHeartPulmonary.h
+++ b/src/model/ClosedLoopHeartPulmonary.h
@@ -37,13 +37,13 @@
 #include "Block.h"
 #include "SparseSystem.h"
 
-// [TODO] get rid of PI.
 /**
- * @brief Define PI
+ * @brief Define math constants (for M_PI)
  *
  */
-#define PI 3.14159265
+#define _USE_MATH_DEFINES
 #include <math.h>
+#include <cmath>
 
 /**
  * @brief Heart and pulmonary circulation model

--- a/src/model/ClosedLoopHeartPulmonary.h
+++ b/src/model/ClosedLoopHeartPulmonary.h
@@ -38,6 +38,10 @@
 #include "SparseSystem.h"
 
 // [TODO] get rid of PI.
+/**
+ * @brief Define PI
+ *
+ */
 #define PI 3.14159265
 #include <math.h>
 

--- a/src/model/ClosedLoopHeartPulmonary.h
+++ b/src/model/ClosedLoopHeartPulmonary.h
@@ -42,7 +42,7 @@
  *
  */
 #define _USE_MATH_DEFINES
-#include <math.h>
+
 #include <cmath>
 
 /**

--- a/src/model/Junction.h
+++ b/src/model/Junction.h
@@ -132,7 +132,15 @@ class Junction : public Block {
    */
   TripletsContributions num_triplets{0, 0, 0};
 
+  /**
+   * @brief Number of inlets to the block.
+   *
+   */
   int num_inlets;
+  /**
+   * @brief Number of outlets from the block.
+   *
+   */
   int num_outlets;
 };
 

--- a/src/model/Model.h
+++ b/src/model/Model.h
@@ -116,7 +116,7 @@ class Model {
   /**
    * @brief Get a block type by its name
    *
-   * @param name The name of the block 
+   * @param name The name of the block
    * @return BlockType The block type
    */
   BlockType get_block_type(const std::string_view &name) const;
@@ -253,7 +253,7 @@ class Model {
   /**
    * @brief Get the number of blocks in the model
    *
-   * @param internal Toggle whether to return internal/hidden blocks 
+   * @param internal Toggle whether to return internal/hidden blocks
    *
    * @return int Number of blocks
    */

--- a/src/model/Model.h
+++ b/src/model/Model.h
@@ -116,7 +116,7 @@ class Model {
   /**
    * @brief Get a block type by its name
    *
-   * @param block_id Global ID of the Block
+   * @param name The name of the block 
    * @return BlockType The block type
    */
   BlockType get_block_type(const std::string_view &name) const;
@@ -134,6 +134,7 @@ class Model {
    *
    * @param inlet_eles Inlet blocks of the node
    * @param outlet_eles Outlet blocks of the node
+   * @param name Name of node
    * @return int Global ID of the node
    */
   int add_node(const std::vector<Block *> &inlet_eles,
@@ -251,6 +252,8 @@ class Model {
 
   /**
    * @brief Get the number of blocks in the model
+   *
+   * @param internal Toggle whether to return internal/hidden blocks 
    *
    * @return int Number of blocks
    */

--- a/src/model/Node.h
+++ b/src/model/Node.h
@@ -81,7 +81,7 @@ class Node {
   /**
    * @brief Set up the degrees of freedom (DOF) of the block
    *
-   * Set \ref global_var_ids and \ref global_eqn_ids of the element based on the
+   * Set global_var_ids and global_eqn_ids of the element based on the
    * number of equations and the number of internal variables of the
    * element.
    *

--- a/src/model/Parameter.h
+++ b/src/model/Parameter.h
@@ -64,6 +64,7 @@ class Parameter {
    * @param id Global ID of the parameter
    * @param times Time steps corresponding to the time-dependent values
    * @param values Values corresponding to the time steps
+   * @param periodic Is this parameter periodic with a cardiac cycle?
    */
   Parameter(int id, const std::vector<double>& times,
             const std::vector<double>& values, bool periodic = true);

--- a/src/optimize/calibrate.h
+++ b/src/optimize/calibrate.h
@@ -45,6 +45,8 @@
 
 /**
  * @brief Main function to run the 0D model calibration.
+ * @param config JSON configuration for 0D model
+ * @return Calibrated JSON configuration for the 0D model
  */
 nlohmann::json calibrate(const nlohmann::json &config);
 

--- a/src/optimize/calibrate.h
+++ b/src/optimize/calibrate.h
@@ -43,6 +43,9 @@
 #include "Model.h"
 #include "debug.h"
 
+/**
+ * @brief Main function to run the 0D model calibration.
+ */
 nlohmann::json calibrate(const nlohmann::json &config);
 
 #endif  // SVZERODSOLVER_OPTIMIZE_CALIBRATOR_HPP_

--- a/src/solve/csv_writer.cpp
+++ b/src/solve/csv_writer.cpp
@@ -38,6 +38,7 @@
  * @param model The underlying model
  * @param mean Toggle whether only the mean over all time steps should be
  * written
+ * @param derivative Toggle whether to output time-derivatives
  * @return CSV encoded output string
  */
 std::string to_vessel_csv(const std::vector<double> &times,
@@ -169,6 +170,7 @@ std::string to_vessel_csv(const std::vector<double> &times,
  * @param model The underlying model
  * @param mean Toggle whether only the mean over all time steps should be
  * written
+ * @param derivative Toggle whether to output time-derivatives
  * @return CSV encoded output string
  */
 std::string to_variable_csv(const std::vector<double> &times,

--- a/src/solve/debug.h
+++ b/src/solve/debug.h
@@ -36,6 +36,9 @@
 
 #include <iostream>
 
+/**
+ * @brief DEBUG_MSG Macro to print debug messages for debug build
+ */
 #ifndef NDEBUG
 #define DEBUG_MSG(str)                                   \
   do {                                                   \

